### PR TITLE
Initial stack state implementation

### PR DESCRIPTION
**Pull request by bquenin**
Relates to #162 
## Verification

```
$ sudo ./swarm start
$ make test-stack
```

### CODE REVIEW

File: api/rpc/stack/stack-state.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+	stack := &Stack{}
+	err := runtime.Store.Get(context.Background(), "stacks/"+s.Id, stack, false)
+	if err != nil {
+		return -1, err
+	}
+	return stack.State, nil
+}
+
+func (s *Stack) setState(state int32) error {
+	stack := &Stack{}
```
Comment (subfuzion) [inline api/rpc/stack/stack-state.go:66]: Technically we will need to do get/set atomically (part of a transaction). You don't want the state changing during the processing of this request. We can add it later, but please leave a TODO comment. (This is a general comment that applies to `transitionTo`, etc).

File: api/rpc/stack/stack-state.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+	Starting
+	// Running  state
+	Running
+	// Redeploying  state
+	Redeploying
+)
+
+var stackRuleSet = RuleSet{
+	//   | Stopped   | Starting  | Running   | Redeploying
+	[]bool{false /**/, true /* */, false /**/, true /* */}, // Stopped (initial state)
```
Comment (generalhenry) [inline api/rpc/stack/stack-state.go:23]: looks like they should be arrays eg: `[4]bool{ false, true, false, true },`

Comment (bquenin) [reply api/rpc/stack/stack-state.go:23]: I'm not sure why slice is not enough ?

Comment (generalhenry) [reply api/rpc/stack/stack-state.go:23]: Slice will get the job done, array just makes it clearer.

Comment (subfuzion) [reply api/rpc/stack/stack-state.go:23]: If you want an array literal, you can let the compiler do the counting: `[...]bool`, but i think a slice literal is fine unless we want to explicitly fix the size.

File: api/rpc/stack/stack-state.go (hunk 1 above)
Comment (subfuzion) [reply api/rpc/stack/stack-state.go:66]: The rest LGTM. Will merge after a comment is added.

**Comment by subfuzion** [comment]
Nice work 👍

**Review by freignat91 — COMMENTED** [review]
Need to use last version of service.ServiceSpec when available instead of stack.Service

**Review by subfuzion — APPROVED** [review]
Nice work 👍

**Pull request merged**